src/team_state.py

```diff
@@ -90,14 +142,41 @@ def from_config(cls, team_state: Dict[str, Any]):
             outs_for_inning,
             lineup,
             starting_pitcher,
-            stats,
+            stlats,
+            game_stats,
             blood,
+            cur_battter_pos,
         )
 
     def get_batter_stats_by_position(self, position: int) -> Tuple[str, Dict[str, float]]:
         batter = self.lineup[position]
         stats = self.stats[batter]
         return batter, stats
 
-    def get_player_stats_by_id(self, id: str) -> Dict[str, float]:
-        return self.stats[id]
+    def get_player_stats_by_id(self, player_id: str) -> Dict[str, float]:
+        return self.stats[player_id]
+
+    def next_batter(self) -> None:
+        if len(self.lineup) == self.cur_batter:
```

Comment:
this is comparing the length of the lineup against the id of the current batter, needs to either lookup the batter at the last lineup position or compare the length against cur_batter_pos

Reply:
Fixed